include/CityExplorer.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <map>
+#include <string>
+#include <algorithm>
+#include <iterator>
+#include <cmath>
+#include <stdexcept>
+#include "Coordinate.h"
+
+// Distance function type
+using DistanceFunction = double (*)(const Coordinate&, const Coordinate&);
```

Comment:
there is std::function in the STL library that you can use and not a pointer to function

Reply:
now using it